Content/Items/Dungeon/Accessories.StoneOfTheDrowned.cs

```diff
@@ -0,0 +1,102 @@
+﻿using ReLogic.Content;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Jungle;
+using StarlightRiver.Core.Systems.BarrierSystem;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Terraria.DataStructures;
+using Terraria.GameContent;
+using Terraria.ID;
+
+namespace StarlightRiver.Content.Items.Dungeon
+{
+	public class StoneOfTheDrowned : SmartAccessory
+	{
+		internal float slotsUsed; // used so you only summon the amount of slots just gained. Otherwise would populate all of the players remaining slots
+		// i love private terraria methods which require reflection to access
+		public static MethodInfo? playerItemCheckShoot_Info;
+		public static Action<Player, int, Item, int>? playerItemCheckShoot;
+
+		public override string Texture => AssetDirectory.JungleItem + Name;
+
+		public StoneOfTheDrowned() : base("Stone of the Drowned", "+30 barrier\nIncreases your number of max minions by 2 when you have no barrier\nAutomatically summons two slots worth of minions if there is a valid weapon in your inventory") { }
```

Comment:
lets shorten this to something like:

Increases your max number of minions by 2 while you have no barrier
Automatically re-summons 2 minions when you reach 0 barrier